### PR TITLE
Add support for code actions on modeline

### PR DESCRIPTION
**Pull request by ericdallo**
This PR adds a new UI for code actions via modeline.

I didn't change the default UI (sideline), so to enable modeline UI, you need to `(setq lsp-ui-code-actions-type 'modeline)`.

* Single code action:
image
* Multiple code actions:
image
* Fallback to unicode if `all-the-icons` is not installed:
image

Currently, all the core logic of code-actions is on `lsp-sideline.el`, that's not good as `sideline` is a UI implementation detail, we may need to refactor that on the future externalizing to a `lsp-code-actions.el` or something.

Closes https://github.com/emacs-lsp/lsp-mode/issues/1237

### CODE REVIEW

File: lsp-ui-modeline.el [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                                        'face 'lsp-ui-modeline-code-action-face)))))
+    (propertize string
+                'help-echo (concat "Apply code actions (s-l a a)\nmouse-1: "
+                                   (if single-action?
+                                       first-action-string
+                                     "select from multiple code actions"))
+                'mouse-face 'mode-line-highlight
+                'local-map (let ((map (make-sparse-keymap)))
+                             (define-key map [down-mouse-1]
+                               #'lsp-ui-sideline-apply-code-actions)
```
Comment (ericdallo) [inline lsp-ui-modeline.el:87]: @yyoncho For some weird reason, the keymap is not working when added to modeline, on any other string this seems to work.
Could you help me with that?

**Comment by ericdallo** [comment]
c/c @matklad

**Comment by yyoncho** [comment]
The idea of emacs-lsp/lsp-mode#1237 (better described in https://github.com/emacs-lsp/lsp-mode/issues/1772) is to make lsp-mode self-contained and provide that feature in lsp-mode, not in lsp-ui.

File: lsp-ui-modeline.el (hunk 1 above)
Comment (yyoncho) [reply lsp-ui-modeline.el:87]: check lsp--download-status - you should create map using make-mode-line-mouse-map

**Comment by ericdallo** [comment]
Move support to lsp-mode: https://github.com/emacs-lsp/lsp-mode/pull/1781, we should open another PR on `lsp-ui` overriding the modeline if sideline is enabled.

**Pull request closed without merging**